SilverSolver/assignment4/tests.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+from islands import *
+
+class TestIslandFill(unittest.TestCase):
+
+    def test_0(self):
+        self.assertEqual(island_fill(0, 0, [[0]]), [[0]])
+        self.assertEqual(island_fill(0, 0, [[1]]), [[2]])
+
+    def test_1(self):
+        self.assertEqual(island_fill(0, 0, [[1, 1], [1, 0]]), [[2, 2], [2, 0]])
+        self.assertEqual(island_fill(1, 1, [[1, 1], [1, 0]]), [[1, 1], [1, 0]])
+
+class TestIslandsNum(unittest.TestCase):
+
+    def test_0(self):
+        self.assertEqual(get_number_of_islands([[0]]), 0)
+        self.assertEqual(get_number_of_islands([[1]]), 1)
+
+    def test_1(self):
+        self.assertEqual(get_number_of_islands([[1,0,1,0]]), 2)
+
+    def test_2(self):
+        self.assertEqual(get_number_of_islands([[1,0,1,0],[0,1,1,1],[0,0,1,0]]), 2)
+        self.assertEqual(get_number_of_islands( \
+                [[1,0,1,0],[0,1,1,1],[0,0,1,0],[1,1,0,0],[0,1,0,1]]), 4)
```

Comment:
Please align the examples vertically, so that we can vizualize the islands better :)
You can also use strings that get_number_of_islands transforms into a grid.
This is also equivalent to the deepcopy you currently use in terms of performance.

self.assertEqual(
    get_number_of_islands(
        '01010',
        '00111',
        '10010',
        '01100',
        '10101'), 6)

Well, I think this example show that the algorithm is incorrect: I count 8 islands here. 

I think a small counter example is:

self.assertEqual(
    get_number_of_islands(
        '01',
        '01'), 1)

if this works please add the unit test:

self.assertEqual(
    get_number_of_islands(
        '010',
        '010',
        '010'), 1)